scrapy/core/scheduler.py

```diff
@@ -229,19 +287,55 @@ def close(self, reason: str) -> Optional[Deferred]:
             self._write_dqs_state(self.dqdir, state)
         return self.df.close(reason)
 
+    def _enqueue_request(self, request: Request) -> None:
+        dqok = self._dqpush(request)
+        if dqok:
+            self.stats.inc_value("scheduler/enqueued/disk", spider=self.spider)
+        else:
+            self._mqpush(request)
+            self.stats.inc_value("scheduler/enqueued/memory", spider=self.spider)
+
```

Comment:
`self._dqpush(request)` called twice per produced request: on `_enqueue_request` and on later line 335  on `enqueue_request`. So It will add 2 requests per 1 req. received from callback.
It return error on executing following scrapy as script application (w/o custom delays)
<details><summary>test1.py</summary>

```python
import scrapy; from scrapy.crawler import CrawlerProcess

class QuotesToScrapeSpider(scrapy.Spider):
    name = "quotes"; custom_settings = {"DOWNLOAD_DELAY": 1}
    def start_requests(self):
        yield scrapy.Request(url='http://quotes.toscrape.com/', callback=self.parse)

    def parse(self, response):
        yield {'w': response.url[-4:]}
        if next_page := response.css("li.next a::attr(href)").get():
            yield scrapy.Request(url=response.urljoin(next_page), callback=self.parse)

if __name__ == "__main__":
    process = CrawlerProcess(); process.crawl(QuotesToScrapeSpider); process.start()
```

</details>

<details><summary>log_output(fragments)</summary>

```
2023-02-24 23:27:18 [scrapy.core.engine] INFO: Error while removing request from slot
Traceback (most recent call last):
  File "C:\Users\georg\anaconda3\envs\scrapy_dev\lib\site-packages\twisted\internet\defer.py", line 857, in _runCallbacks
    current.result = callback(  # type: ignore[misc]
  File "C:\Users\georg\PycharmProjects\scrapy\scrapy\core\engine.py", line 204, in <lambda>
    d.addBoth(lambda _: self.slot.remove_request(request))
  File "C:\Users\georg\PycharmProjects\scrapy\scrapy\core\engine.py", line 54, in remove_request
    self.inprogress.remove(request)
KeyError: <GET http://quotes.toscrape.com/page/9/>
2023-02-24 23:27:19 [scrapy.core.engine] DEBUG: Crawled (200) <GET http://quotes.toscrape.com/page/10/> (referer: http://quotes.toscrape.com/page/9/)
2023-02-24 23:27:19 [scrapy.core.scraper] DEBUG: Scraped from <200 http://quotes.toscrape.com/page/10/>
{'w': '/10/'}
2023-02-24 23:27:19 [scrapy.core.engine] INFO: Closing spider (finished)
2023-02-24 23:27:19 [scrapy.statscollectors] INFO: Dumping Scrapy stats:
{'downloader/exception_count': 9,
 'downloader/exception_type_count/builtins.KeyError': 9,
 'downloader/request_bytes': 5384,
 'downloader/request_count': 20,
 'downloader/request_method_count/GET': 20,
 'downloader/response_bytes': 21915,
 'downloader/response_count': 10,
 'downloader/response_status_count/200': 10,
 'elapsed_time_seconds': 23.200037,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2023, 2, 24, 21, 27, 19, 523219),
 'httpcompression/response_bytes': 108561,
 'httpcompression/response_count': 10,
 'item_scraped_count': 10,
 'log_count/DEBUG': 21,
 'log_count/ERROR': 9,
 'log_count/INFO': 19,
 'log_count/WARNING': 1,
 'request_depth_max': 9,
 'response_received_count': 10,
 'scheduler/dequeued': 20,
 'scheduler/dequeued/memory': 20,
 'scheduler/enqueued': 20,
 'scheduler/enqueued/memory': 20,
 'start_time': datetime.datetime(2023, 2, 24, 21, 26, 56, 323182)}
2023-02-24 23:27:19 [scrapy.core.engine] INFO: Spider closed (finished)

Process finished with exit code 0

```

</details>

It leads to inconsistency between ` 'scheduler/enqueued': 20`, and  ` 'downloader/response_count': 10,` in result stats

Reply:
Recent updates resolves this issue. Now it works.. as expected